From fdc069542d03443bb095beefa22ce819dc843b0b Mon Sep 17 00:00:00 2001
From: Yandong Lin <yandong.lin@rock-chips.com>
Date: Thu, 4 Feb 2021 17:20:23 +0800
Subject: [PATCH] video: rockchip: mpp: iep2: remove reduce_req function

1. Because of hw problems before, there will be problems with
high-frequency reset. So the frequency must be reduced before reset.
But this operation is unnecessary for now.
2. Iep share clk with rga on rk356x, so frequency reduction may affect rga.

Signed-off-by: Yandong Lin <yandong.lin@rock-chips.com>
Change-Id: I469d2d6ba2671560acd9eaf6b86d57411c7e3de5
---
 drivers/video/rockchip/mpp/mpp_iep2.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_iep2.c b/drivers/video/rockchip/mpp/mpp_iep2.c
index 3f38180498d9..14f514a70cce 100644
--- a/drivers/video/rockchip/mpp/mpp_iep2.c
+++ b/drivers/video/rockchip/mpp/mpp_iep2.c
@@ -854,15 +854,6 @@ static int iep2_set_freq(struct mpp_dev *mpp,
 	return 0;
 }
 
-static int iep2_reduce_freq(struct mpp_dev *mpp)
-{
-	struct iep2_dev *iep = to_iep2_dev(mpp);
-
-	mpp_clk_set_rate(&iep->aclk_info, CLK_MODE_REDUCE);
-
-	return 0;
-}
-
 static int iep2_reset(struct mpp_dev *mpp)
 {
 	struct iep2_dev *iep = to_iep2_dev(mpp);
@@ -888,7 +879,6 @@ static struct mpp_hw_ops iep_v2_hw_ops = {
 	.clk_on = iep2_clk_on,
 	.clk_off = iep2_clk_off,
 	.set_freq = iep2_set_freq,
-	.reduce_freq = iep2_reduce_freq,
 	.reset = iep2_reset,
 };
 
-- 
2.35.3

